lib/jsonwp-proxy/proxy.js

```diff
@@ -232,15 +236,111 @@ class JWProxy {
     }
   }
 
+  async getDownstreamProtocol () {
+    if (!this._downstreamProtocol) {
+      const [, resBody] = await this.proxy('/status', 'GET');
+      this._downstreamProtocol = this.getProtocolFromResBody(resBody);
+    }
+    return this._downstreamProtocol;
+  }
+
+  requestToCommandName (url, method) {
+    const extractCommandName = (pattern) => {
+      const pathMatch = pattern.exec(url);
+      return pathMatch ? routeToCommandName(pathMatch[1], method) : null;
+    };
+    let commandName = routeToCommandName(url, method);
+    if (!commandName && _.includes(url, '/wd/hub/session/')) {
```

Comment:
what if we move to urls which don't include `/wd/hub` as a prefix? this isn't part of the spec so we could conceivably move to no prefix, or configurable prefix

Reply:
`routeToCommandName` call itself should cover such case. These two conditions were added just for the sake of safety.